### PR TITLE
Employee candidate table combine

### PR DESCRIPTION
**Pull request by Wimmas**
Removed Candidate table. Moved all functionality to Employee model and added fields to Employee table to accommodate the change.

image

protected override void Up(MigrationBuilder migrationBuilder)
{
migrationBuilder.DropTable(
name: "Candidate");

 migrationBuilder.AddColumn<string>(
     name: "blackListReason",
     table: "Employee",
     type: "text",
     nullable: true);

 migrationBuilder.AddColumn<int>(
     name: "blackListStatus",
     table: "Employee",
     type: "integer",
     nullable: true);

 migrationBuilder.AddColumn<string>(
     name: "cv",
     table: "Employee",
     type: "text",
     nullable: true);

 migrationBuilder.AddColumn<string>(
     name: "highestQualification",
     table: "Employee",
     type: "text",
     nullable: true);

 migrationBuilder.AddColumn<bool>(
     name: "isCandidate",
     table: "Employee",
     type: "boolean",
     nullable: false,
     defaultValue: false);

 migrationBuilder.AddColumn<string>(
     name: "linkedIn",
     table: "Employee",
     type: "text",
     nullable: true);

 migrationBuilder.AddColumn<string>(
     name: "portfolioLink",
     table: "Employee",
     type: "text",
     nullable: true);

 migrationBuilder.AddColumn<string>(
     name: "portfolioPdf",
     table: "Employee",
     type: "text",
     nullable: true);

 migrationBuilder.AddColumn<int>(
     name: "qualificationEndDate",
     table: "Employee",
     type: "integer",
     nullable: true);

 migrationBuilder.AddColumn<int>(
     name: "referal",
     table: "Employee",
     type: "integer",
     nullable: true);

 migrationBuilder.AddColumn<string>(
     name: "school",
     table: "Employee",
     type: "text",
     nullable: true);
}

### CODE REVIEW

File: RR.UnitOfWork/Entities/HRIS/Employee.cs [hunk 1]
```diff
@@ -57,6 +58,45 @@ public Employee(EmployeeDto employeeDto, EmployeeTypeDto employeeType)
...
+        PortfolioPDF = employeeDto.PortfolioPDF;
+        Referral = employeeDto.Referral;
+        HighestQualification = employeeDto.HighestQualification;
+        School = employeeDto.School;
+        QualificationEndDate = employeeDto.QualificationEndDate;
+        BlackListStatus = employeeDto.BlackListStatus;
+        BlackListReason = employeeDto.BlackListReason;
+        IsCandidate = employeeDto.IsCandidate;
+    }
+
```
Comment (tbddorothy) [inline RR.UnitOfWork/Entities/HRIS/Employee.cs:74]: Shouldn't both of these constructors use EmployeeDto? also there's quite some duplication

Comment (Wimmas) [reply RR.UnitOfWork/Entities/HRIS/Employee.cs:74]: The reason they don't is that the second one is for the Canidate replacement. I basically still make use of the candidate service and dto, but use the Employee model. So I've added the second constructor for use from the candidate service.

**Comment by EduanO** [comment]
I had a discussion with @tbddorothy  in regards to this. I do not agree to this approach as it is not the right approach for separation of responsibility.

I get it contains the same data to some degree but it is essentially 2 different system data for different actions.

**Comment by tbddorothy** [comment]
> I had a discussion with @tbddorothy in regards to this. I do not agree to this approach as it is not the right approach for separation of responsibility.
> 
> I get it contains the same data to some degree but it is essentially 2 different system data for different actions.

I'm closing is PR

**Pull request closed without merging**